editor/src/components/inspector/constraints-section.tsx

```diff
@@ -0,0 +1,259 @@
+/** @jsxRuntime classic */
+/** @jsx jsx */
+import { jsx } from '@emotion/react'
+
+import React from 'react'
+import { useContextSelector } from 'use-context-selector'
+import type { LayoutPinnedPropIncludingCenter } from '../../core/layout/layout-helpers-new'
+import { when } from '../../utils/react-conditionals'
+import { FlexColumn, FlexRow, InspectorSubsectionHeader, PopupList, UtopiaTheme } from '../../uuiui'
+import { getControlStyles } from '../../uuiui-deps'
+import { InspectorRowHoverCSS } from '../context-menu-wrapper'
+import { useDispatch } from '../editor/store/dispatch-context'
+import { Substores, useEditorState, useRefEditorState } from '../editor/store/store-hook'
+import type { FramePinsInfo } from './common/layout-property-path-hooks'
+import { InspectorPropsContext } from './common/property-path-hooks'
+import { PinControl } from './controls/pin-control'
+import {
+  GroupChildPinControl,
+  GroupConstraintSelect,
+  allElementsAreGroupChildren,
+  anySelectedElementGroupOrChildOfGroup,
+} from './fill-hug-fixed-control'
+import { selectedViewsSelector } from './inpector-selectors'
+import type { RequestedPins } from './simplified-pinning-helpers'
+import {
+  HorizontalPinChangeOptions,
+  HorizontalPinChangeOptionsIncludingMixed,
+  VerticalPinChangeOptions,
+  VerticalPinChangeOptionsIncludingMixed,
+  getFixedPointsForPinning,
+  getFrameChangeActions,
+  useDetectedPinning,
+} from './simplified-pinning-helpers'
+import { PinHeightSVG, PinWidthSVG } from './utility-controls/pin-control'
+import { UIGridRow } from './widgets/ui-grid-row'
+
+export const ConstraintsSection = React.memo(() => {
+  const noGroupOrGroupChildrenSelected = !useEditorState(
+    Substores.metadata,
+    anySelectedElementGroupOrChildOfGroup,
+    'ConstraintsSection someGroupOrGroupChildrenSelected',
+  )
+  const onlyGroupChildrenSelected = useEditorState(
+    Substores.metadata,
+    allElementsAreGroupChildren,
+    'ConstraintsSection onlyGroupChildrenSelected',
+  )
+
+  return (
+    <React.Fragment>
+      <InspectorSubsectionHeader>
+        <FlexRow
+          style={{
+            flexGrow: 1,
+            height: 42,
+          }}
+        >
+          <span style={{ flex: 1 }}>Constraints</span>
+        </FlexRow>
+      </InspectorSubsectionHeader>
+      {when(noGroupOrGroupChildrenSelected, <FrameChildConstraintsSection />)}
+      {when(onlyGroupChildrenSelected, <GroupChildConstraintsSection />)}
+    </React.Fragment>
+  )
+})
+ConstraintsSection.displayName = 'ConstraintsSection'
+
+const GroupChildConstraintsSection = React.memo(() => {
+  return (
+    <FlexColumn css={{ paddingBottom: UtopiaTheme.layout.rowHorizontalPadding }}>
+      <UIGridRow padded variant='<-auto-><----------1fr--------->'>
+        <GroupChildPinControl />
+        <FlexColumn css={{ gap: 0 }}>
```

Comment:
imho this should match the height of the pin control

Reply:
fixed!